device/globals/global_vm.gd

```diff
@@ -87,40 +87,11 @@ func settings_loaded(p_settings):
 	AudioServer.set_bus_volume_db(0, settings.sfx_volume)
 	TranslationServer.set_locale(settings.text_lang)
 	music_volume_changed()
-	update_window_fullscreen(true)
 	get_tree().call_group_flags(SceneTree.GROUP_CALL_DEFAULT, "ui", "language_changed")
 
-func update_window_fullscreen(p_force = false):
-	if ProjectSettings.get_setting("debug/screen_size_override"):
-		return
-	if !p_force && (settings.fullscreen == OS.is_window_fullscreen()):
-		return
-	if settings.fullscreen:
-		OS.set_window_resizable(true)
-		OS.set_window_fullscreen(settings.fullscreen)
-		pass
-	else:
-		var title = ProjectSettings.get_setting("platform/window_title_height")
-		var sc = OS.get_current_screen()
-		var ratio = 1080 / 1920.0
-		var size = OS.get_screen_size(sc)
-		printt("***** got screen size ", size, title)
-		var h = size.y - title
-		if h / ratio > size.x:
-			size.y = size.x * ratio
-		else:
-			size.y = size.y - title
-			size.x = h / ratio
-		printt("setting window to size", size)
-		OS.set_window_fullscreen(settings.fullscreen)
-		OS.set_window_size(size)
-		#OS.set_window_position(Vector2(0, 0))
-		OS.set_window_resizable(ProjectSettings.get_setting("platform/screen_resizable"))
-
```

Comment:
Did this not work at all?

Reply:
If it did, I don't know what it was supposed to do 😱 

Godot resizes the window just fine, there's no need to go out to `OS` for it that I can see.

A valid use case for resizable would be when shrinking the window, but this would involve setting the viewport and I have no idea how to do it. I tried and gave up :( There should also be a sanity check that the window isn't larger than the camera limits or you get the gray background outside.

The least surprising code would be to deny resizing on the Escoria level and figure out a way to change resolutions/scaling through menus or something.

Is there a case for the opposite?

But this code doesn't seem to serve any purpose, though maybe it did in 2.1. @StraToN may be the leading expert in this matter. plz halp!

Reply:
It might be a remnant from an older version of Godot then. Yeah, I think @StraToN should take a look at this before it's merged, just in case.